File: swift/Sources/ProjectNameSDK/core/BluefinTecsMerchantPortalError.swift
// BluefinTecsMerchantPortalError - the SDK error type. Carries the pipeline error code, the
// originating context and cleaned result/spec snapshots.

import Foundation

public final class BluefinTecsMerchantPortalError: Error, CustomStringConvertible {
  public let isBluefinTecsMerchantPortalError = true
  public let sdk = "BluefinTecsMerchantPortal"
  public let code: String
  public let message: String
  public var ctx: Context?
  public var resultVal: Value = .noval
  public var specVal: Value = .noval

  public init(_ code: String, _ message: String, _ ctx: Context?) {
    self.code = code
    self.message = message
    self.ctx = ctx
  }

  public var description: String { message }
}

File: csharp/core/BluefinTecsMerchantPortalError.cs
// BluefinTecsMerchantPortalError - the SDK error type. Carries the pipeline error code,
// the originating context and cleaned result/spec snapshots.

namespace BluefinTecsMerchantPortalSdk;

public class BluefinTecsMerchantPortalError : Exception
{
    public bool IsBluefinTecsMerchantPortalError = true;
    public string Sdk = "BluefinTecsMerchantPortal";
    public string Code;
    public Context? Ctx;
    public object? ResultVal;
    public object? SpecVal;

    public BluefinTecsMerchantPortalError(string code, string msg, Context? ctx)
        : base(msg)
    {
        Code = code;
        Ctx = ctx;
    }
}
